python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -437,18 +452,20 @@ cdef class Column:
         except AttributeError:
             raise TypeError("Object does not implement __array_interface__")
 
-        arr = np.asarray(obj)
+        shape, strides, dtype = _parse_array_interface(iface)
+
         # ravel() is needed here because even though the array
         # is C-contiguous, viewing it as np.uint8 retains its
         # original shape. DeviceBuffer.to_device expects a 1D buffer,
         # so we call ravel() to ensure the data is flattened. This
         # doesn't copy the data for C-contiguous arrays, just creates
         # a flat view.
+        arr = np.asarray(obj)
         flat_data_view = memoryview(arr.ravel().view(np.uint8))
         device_buffer = DeviceBuffer.to_device(flat_data_view)
```

Comment:
I wonder again if we can avoid the use of `numpy`. We're effectively using this as a complicated way to get a C-contiguous `memoryview` of a pointer.

How about:

```
nbytes = product(shape) * itemsize
device_buffer = DeviceBuffer.to_device(<const unsigned char[:nbytes:1]><uintptr_t>data_ptr)
```
Where `data_ptr` is the integer pointer in the array interface description.

No need for numpy at all.

Reply:
Thanks, cython wouldn't let me cast directly to a typed memoryview. See https://github.com/rapidsai/cudf/pull/18425/files#diff-a3c2e9bea60260df24118e4d2f6c63c402e52b23cefe414310ab326fc2e6cde4R472